dev/src/document.ts

```diff
@@ -163,7 +167,7 @@ export class DocumentSnapshot {
    * @param {Object} obj - The object to store in the DocumentSnapshot.
```

Comment:
Are these type annotations still intentional?

Reply:
I only want to keep them for external methods, as they are used in JSDoc. Removed from here.

FYI: document.ts still has room for a bunch of TS-related improvements.